### PR TITLE
Update stringtohglobalansi.cs

### PR DESCRIPTION
**Pull request by Owenll66**
## Summary

Formatted the code and corrected the comment of the result string printed after FreeHGlobal is called.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/dotnet-api-docs?pullRequest=8695) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/dotnet-api-docs?pullRequest=8695) before we can accept your contribution.<br/><br/>:x: Owenll66 [sign now](https://cla.dotnetfoundation.org/dotnet/dotnet-api-docs?pullRequest=8695)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/dotnet-api-docs?pullRequest=8695) it.</sub>

File: snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
-    // IntPtr handle value is still the same:
-    Console.WriteLine("4) stringPointer = " + stringPointer );
-
-    // However, it contains no data after being freed:
-    String RetrievedString2 = Marshal.PtrToStringAnsi( stringPointer);
-    Console.WriteLine("5) RetrievedString2 = " + RetrievedString2 );
+        // IntPtr handle value is still the same:
+        Console.WriteLine("4) stringPointer = " + stringPointer);
+        
+        // However, the data may be cleared after being freed. Depending on whether the memory allocated to stringPointer
```
Comment (gewarren) [inline snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs:30]:
```suggestion
        // However, the data may be cleared after the memory is freed, depending on whether the memory allocated to stringPointer
```

File: snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
+        // has been reclaimed or not. Uncomment Thread.Sleep(1000) below will increase the likelihood of the memory been reclaimed.
```
Comment (gewarren) [inline snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs:31]:
```suggestion
        // has been reclaimed or not. Uncommenting the following line (Thread.Sleep(1000)) increases the likelihood of the memory being reclaimed.
```

File: snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs (hunk 1 above)
Comment (Owenll66) [reply snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs]: @gewarren  Thanks for improving the comment, I have combined your change into mine as a single commit. Cheers!

File: snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs (hunk 2 above)
Comment (Owenll66) [reply snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs]: That looks better thanks

File: snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs [hunk 3]
```diff
@@ -1,34 +1,37 @@
...
 
-    // Always free the unmanaged string.
-    Marshal.FreeHGlobal(stringPointer);
+        // Always free the unmanaged string.
+        Marshal.FreeHGlobal(stringPointer);
 
-    // IntPtr handle value is still the same:
-    Console.WriteLine("4) stringPointer = " + stringPointer );
+        // IntPtr handle value is still the same:
+        Console.WriteLine("4) stringPointer = " + stringPointer);
```
Comment (adegeo) [inline snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs:28]: I'm really confused as to how this code can compile?

Comment (Owenll66) [reply snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs:28]: Sorry, I didn't get what you mean. Do you mean it was supposed to have compilation errors on incorrect formats/styles? Possibly a style config issue?

Comment (adegeo) [reply snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs:28]: hahaha no never mind. I didn't notice that the `"` was missing, so I thought the code was `Console.WriteLine("4") other-code` which would be invalid. lol

Comment (Owenll66) [reply snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs:28]: All good, thanks

**Review by elinor-fung — APPROVED** [review]
Thanks!
Sorry I missed seeing this months ago.

**Comment by Owenll66** [comment]
> @Owenll66 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
>  > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree

**Pull request merged**